src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppInstanceStatus.java

```diff
@@ -175,6 +178,16 @@ private DeploymentState mapState() {
 		}
 		return result;
 	}
+
+	private String determineActuatorPort(Pod pod) {
+		return pod.getSpec().getContainers().stream().filter( (Container container) ->
+				container.getLivenessProbe() != null &&
+				container.getLivenessProbe().getHttpGet() != null &&
```

Comment:
I ❤️  using `map` as it will handle nulls for us. Since we ultimately want to operation on the `HttpGetAction` it works nicely w/ something like this: 
```java
return pod.getSpec().getContainers().stream()
		.map(Container::getLivenessProbe)
		.map(Probe::getHttpGet)
		.filter((httpGetAction) -> httpGetAction.getPath().startsWith("/actuator"))
		.findFirst()
		.map(HTTPGetAction::getPort)
		.map(IntOrString::getStrVal)
		.orElse(new IntOrString("unknown").getStrVal());
```